File: app/src/main/java/com/example/mvc_example/Controller.java
package com.example.mvc_example;

import com.example.mvc_example.model.Model;

public class Controller  {

    private Model model;
    private MainActivity view;

    public Controller(Model model, MainActivity view) {
        this.model = model;
        this.view = view;
    }

    public void makeApiCall() {
        final String urlEndpoint="apis/get_book.php";

        //url = http://johncastle.com.mx/
        model.execute(BuildConfig.url+urlEndpoint,new Model.OnResult() {
                    @Override
                    public void onSuccess(String result) {
                        view.printResponse(result);
                    }
                    @Override
                    public void onError(Exception error) {
                      prepareError(error);
                    }
                }
        );
    }

    public void prepareError(Exception error){
        view.printError(error.toString());
    }

}

File: app/src/main/java/com/example/mvc_example/Controller.kt
package com.example.mvc_example


class Controller(val model: Model, val view : MainActivityKotlin ) {

    fun makeApiCall() {
        val urlEndpoint = "apis/get_book.php"
        //url = http://johncastle.com.mx/
        model.execute(BuildConfig.url + urlEndpoint, object : Model.OnResult {
            override fun onSuccess(result: String?) {
                view.printResponse(result)
            }

            override fun onError(error: Exception?) {
                if (error != null) {
                    prepareError(error)
                }
            }

        }
        )
    }

    fun prepareError(error: Exception) {
        view.printError(error.toString())
    }

}